org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/job/RulesConfigurationPageSaveJob.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2024 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.ui.internal.job;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.sonarlint.eclipse.core.internal.preferences.SonarLintGlobalConfiguration;
+import org.sonarlint.eclipse.core.internal.resources.ProjectsProviderUtils;
+import org.sonarlint.eclipse.core.internal.telemetry.LinkTelemetry;
+import org.sonarlint.eclipse.ui.internal.util.MessageDialogUtils;
+
+/** Job triggered when the rules configuration page changes are saved in order to not block the UI */
+public class RulesConfigurationPageSaveJob extends Job {
+  public RulesConfigurationPageSaveJob() {
+    super("Saving the rules configuration");
+  }
+
+  @Override
+  protected IStatus run(IProgressMonitor monitor) {
+    var boundToAllProjectsRatio = ProjectsProviderUtils.boundToAllProjectsRatio();
+    if (boundToAllProjectsRatio == 1f) {
```

Comment:
You see, here, I think you could save the comment next line by having code like:
```suggestion
    if (projectsBindingRatio == ALL_BOUND) {
```

Reply:
Done